docs/backends/lua2.rst

```diff
@@ -0,0 +1,173 @@
+ - int domain_id - Associated domain ID
+
+OUTPUT:
+ Same as ``lookup`` function. Return false if not found or wanted.
+
+NOTES:
+ This function is **optional**.
+ 
+``dns_get_domaininfo(domain)``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Get domain information for given domain.
+
+INPUT:
+ - DNSName domain - Domain to get info for
+
+OUTPUT:
+ Return false if not supported or found, otherwise expects a table of:
+ - string account - Associated account of this domain (default: <empty>)
+ - string kind - Domain kind (NATIVE,MASTER,SLAVE) (default: master)
+ - int id - Associated domain ID (default: -1)
+ - int last_check - UNIX timestamp of last check from master
+ - table of strings masters - Master servers for this domain (default: <empty>)
+ - long notified_serial - Notified serial to slaves
+ - long serial - Current domain serial
+
+NOTES:
+ This function is **optional**.
+ masters, account, last_check, notified_serial are for master/slave interaction only.
+ If this function is missing, it will revert into looking up SOA record for the given domain,
+ and uses that, if found.
+
+``dns_get_all_domains()``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Get domain information for all domains.
+
+OUTPUT:
+ Return false if not supported or found, otherwise return a table of string, domaininfo. See ``dns_get_domaininfo```.
+
+NOTES:
+ This function is **optional**, except if you need master functionality.
+
+``dns_get_domain_metadata(domain, kind)``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Get metadata value(s) for given domain and metadata kind.
+
+INPUT:
+ - DNSName domain - Domain to get metadata for
+ - string kind - What kind of metadata to return
+
+OUTPUT:
+ - table of strings. Or false if not supported or found.
+
+NOTES:
+ This function is **required** if ``dns_dnssec`` is true.
+
+``dns_get_all_domain_metadata(domain)``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Get all metadata for domain.
+
+INPUT:
+ - DNSName domain - Domain to get metadata for
+
+OUTPUT:
+ - table of string keys with table of strings. Or false if not supported or found.
+
+NOTES:
+ This function is **optional**.
+
+``dns_get_domain_keys(domain)``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Get DNSSEC key(s) for the given domain. Content must be valid key record in format that PowerDNS understands.
+
+INPUT:
+ - DNSName domain - Domain to get key(s) for
+
+OUTPUT:
+ Return false if not found or supported, otherwise expects table of:
+ - int id - Key ID
+ - int flags - Key flags
+ - bool active - Is key active
+ - string content - Key itself
+
+NOTES:
+ This function is **optional**. However, not implementing this means you cannot do live signing.
+
+``dns_get_before_and_after_names_absolute(id, qname)``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Calculate NSEC before/after value for the given qname for domain with id.
+
+INPUT:
+ - int id - Associated domain id
+ - DNSName qname - DNS name to calculate
+
+OUTPUT:
+ Table of three DNSNames, in order of Unhashed, before and after.
```

Comment:
Regarding some previous comments, I see from the testing script that some functions indeed expect keyed tables. This one does not - suggest saying 'Array of three DNSNames' for clarity.